server/cp/ultrad.py

```diff
@@ -0,0 +1,135 @@
+
+import logging
+import requests
+
+from flask import current_app as app
+from urllib.parse import urljoin
+
+from apps.tasks import send_to
+from superdesk import get_resource_service
+from superdesk.lock import lock, unlock, touch
+from superdesk.text_utils import get_text
+from superdesk.celery_app import celery
+from superdesk.editor_utils import Editor3Content
+
+
+logger = logging.getLogger(__name__)
+
+ULTRAD_ID = 'ultrad_id'
+ULTRAD_URL = 'https://pc-trad.herokuapp.com/cms/'
+ULTRAD_TIMEOUT = (5, 10)
+
+
+class UltradException(RuntimeError):
+    pass
+
+
+def get_headers():
+    return {'x-ultrad-auth': app.config['ULTRAD_AUTH']}
+
+
+def upload_document(item):
+    payload = {
+        'lang': {
+            'fromLang': 'en',
+            'toLang': 'fr',
+        },
+        'name': item['headline'],
+        'state': 'new',
+        'text': {
+            'original': get_text(item['body_html']),
+        },
+    }
+
+    try:
+        resp = requests.post(ULTRAD_URL, json=payload, headers=get_headers(), timeout=ULTRAD_TIMEOUT)
+        resp.raise_for_status()
+    except requests.HTTPError:
+        logger.exception('got error on post %s', item['guid'])
```

Comment:
it would be good to have HTTP code + text in logs, to make debugging easier.

Reply:
you would see the code in the exception info, but the api actually responds 200 also for validation errors and missing auth, so this can be only network issue

Reply:
what about internal error, the API would then return a 500 right?
If this is only network issue, maybe the error message could be more explicit about that.